aldryn_client/cloud.py

```diff
@@ -85,6 +89,19 @@ def upload_boilerplate(self, archive_obj):
         )
         return request.request()
 
+    def download_db(self, website_slug):
+        request = api_requests.DownloadDBRequest(
+            self.session,
+            url_kwargs={'website_slug': website_slug},
+        )
+        temp_dir = create_temp_dir()
+        response = request.request()
+        with tarfile.open(
+                fileobj=StringIO(response.content), mode='r:gz'
```

Comment:
Also note that `pg_dump` and `pg_restore` support dumping/restoring to/from tar archives directly, no need to extract.